lib/open_project/translations/models/i18n_provider.rb

```diff
@@ -119,7 +147,26 @@ def create_temp_file(filename)
     tempfile
   end
 
-  def unlink_temp_file(tempfile)
-    tempfile.unlink
+  def close_tmp_file(tempfile)
+    # this will be garbage collected automatically
+    tempfile.close
+  end
+
+  def upload_old_translations(dest_without_dir, source, language_name, params)
+    locale_temp_file = create_temp_file('old_locale')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.